src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -864,16 +860,19 @@ internal RunVersionOne CreateRun(Run v2Run)
                     _currentRun = run;
 
                     run.Architecture = v2Run.Architecture;
-                    run.AutomationId = v2Run.AutomationLogicalId;
                     run.BaselineId = v2Run.BaselineInstanceGuid;
                     run.Files = v2Run.Files?.ToDictionary(v => v.Key, v => CreateFileData(v.Value));
-                    run.Id = v2Run.InstanceGuid;
+
+                    run.Id = v2Run.Id?.InstanceGuid;
+                    run.AutomationId = v2Run.AggregateIds?.First().InstanceId;
```

Comment:
>First [](start = 59, length = 5)

`FirstOrDefault()?.InstanceId`, on the off chance that the array is present but empty.

Reply:
we actually should be scrubbing for this, nearly every array in the standard should be minValues == 1. will open a tracking issue.

---
In reply to: [226735459](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226735459) [](ancestors = 226735459)

Reply:
You added the "?" but you didn't change `First` to `FirstOrDefault`. `First` will throw if the array is empty. (I know you opened a tracking issue but we have the hood up on this one).

---
In reply to: [226844620](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226844620) [](ancestors = 226844620,226735459)

Reply:
Resolved in subsequent PR.

---
In reply to: [226873071](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226873071) [](ancestors = 226873071,226844620,226735459)